src/aap_eda/services/sync_certs.py

```diff
@@ -0,0 +1,197 @@
+        # remove it from Gateway
+        if existing_object and not inputs["certificate"]:
+            return self.delete()
+
+        # If the user has not provided any certificate nothing to do
+        if not inputs["certificate"]:
+            return
+
+        sha256 = hashlib.sha256(
+            inputs["certificate"].encode("utf-8")
+        ).hexdigest()
+
+        if existing_object.get("sha256", "") != sha256:
+            data = {
+                "name": self.eda_credential.name,
+                "pem_data": inputs["certificate"],
+                "sha256": sha256,
+                "remote_id": self._get_remote_id(),
+            }
+            headers = self._prep_headers()
+            if existing_object:
+                slug = f"{SLUG}/{existing_object['id']}/"
+                url = urljoin(self.gateway_url, slug)
+                response = requests.patch(
+                    url,
+                    json=data,
+                    headers=headers,
+                    verify=self.gateway_ssl_verify,
+                    timeout=DEFAULT_TIMEOUT,
+                )
+            else:
+                url = urljoin(self.gateway_url, SLUG)
+                response = requests.post(
+                    url,
+                    json=data,
+                    headers=headers,
+                    verify=self.gateway_ssl_verify,
+                    timeout=DEFAULT_TIMEOUT,
+                )
+
+            if response.status_code in [
+                status.HTTP_200_OK,
+                status.HTTP_201_CREATED,
+            ]:
+                LOGGER.debug("Certificate updated")
+            elif response.status_code == status.HTTP_400_BAD_REQUEST:
+                LOGGER.error("Update failed")
+            else:
+                LOGGER.error("Couldn't update certificate")
+
+        else:
+            LOGGER.debug("No changes detected")
+
+    def delete(self, event_stream_id: Optional[int]):
+        """Delete the Certificate from Gateway."""
+        existing_object = self._fetch_from_gateway()
+        if not existing_object:
+            return
+
+        objects = models.EventStream.objects.filter(
+            eda_credential_id=self.eda_credential
+        )
+        if not event_stream_id:
+            self._delete_from_gateway(existing_object)
+        elif len(objects) == 1 and event_stream_id == objects[0].id:
+            self._delete_from_gateway(existing_object)
+
+    def _delete_from_gateway(self, existing_object: dict):
+        slug = f"{SLUG}/{existing_object['id']}/"
+        url = urljoin(self.gateway_url, slug)
+        headers = self._prep_headers()
+        response = requests.delete(
+            url,
+            headers=headers,
+            verify=self.gateway_ssl_verify,
+            timeout=DEFAULT_TIMEOUT,
+        )
+        if response.status_code == status.HTTP_200_OK:
+            LOGGER.debug("Certificate object deleted")
+        if response.status_code == status.HTTP_404_NOT_FOUND:
```

Comment:
This should be `elif` otherwise we would be raising GatewayAPIError incorrectly, 
```suggestion
        elif response.status_code == status.HTTP_404_NOT_FOUND:
```